pg_wait_sampling.c

```diff
@@ -75,7 +79,12 @@ get_max_procs_count(void)
 	int count = 0;
 
 	/* MyProcs, including autovacuum workers and launcher */
-	count += MaxBackends;
+	count += MaxConnections + autovacuum_max_workers + 1
+			+ max_worker_processes
+#if PG_VERSION_NUM >= 120000
+			+ max_wal_senders
+#endif
+			+ 1;
```

Comment:
To which kind of process does belong the last one? `MaxBackends` in [PG15](https://github.com/postgres/postgres/blob/002c9dd97a0c874fd1693a570383e2dd38cd40d5/src/backend/utils/init/postinit.c#L555-L557) and before is calculated without this.

Reply:
We don't access to the array for `queryId` values using `MyBackendId` index value. Canonical way to define own index is `MyProc - ProcGlobal->allProcs`

Reply:
Yes, sorry the extra +1 comes from pg_stat_kcache code where we use a 1-based array access.  I wanted to remove it and forgot as I was focusing on the v12 change.  I will fix it!